### ISSUE TITLE
[Twig] Add tag component

### ISSUE THREAD

**Issue by seb-jean** [symfony/ux#300]
Hi,

It would be great to have the tag `component`

```twig
{% component %}
    Content...
{% endcomponent %}
```

Here is an example that @weaverryan and @kbond had made:
https://github.com/weaverryan/twig-components-poc/blob/master/templates/main/index.html.twig#L7

Thanks ;)

**Comment by kbond** [comment]
Agreed, this is on my radar!

**Comment by seb-jean** [comment]
Oh awesome! It's going to be so handy with that!

**Comment by kbond** [comment]
This awesome library is my inspiration: https://github.com/giorgiopogliani/twig-components

"embedded components" is what I've been calling this.

**Comment by seb-jean** [comment]
One of the idea for me is to transcribe this repository https://github.com/seanpdoyle/stimulus_aria_widgets with Twig Component.
For instance https://github.com/seanpdoyle/stimulus_aria_widgets#combobox has 1 helper and 3 targets which generate different attributes.

**Comment by seb-jean** [comment]
@kbond Did you have time to explore the idea? 😄

**Comment by WebMamba** [comment]
Hi guys ! I really like the idea ! And I will be happy to contribute ! So what's your current state on this idea ? How do you wan't to processed, to make it real ? 😁

**Comment by kbond** [comment]
@seb-jean, not yet - was hoping to pick this back up again soon.

@WebMamba, if your willing to get started on this, that would be great. What I was hoping to do was port https://github.com/giorgiopogliani/twig-components over (the slots and embedded component stuff) but with the component class injected (as `this`) just like the current `component()` function.

**Comment by seb-jean** [comment]
Thanks @kbond

**Comment by kbond** [comment]
Started working on this over in #317. Not exactly what I envisioned but the internals of twig are _complex_!

I didn't end up using anything from https://github.com/giorgiopogliani/twig-components. Instead I modeled the feature after twig's native [_embed tag_](https://twig.symfony.com/doc/3.x/tags/embed.html). It's basically an embed tag with access to the component and attributes. The benefit of this is it uses twig's native block tag for the slots. This allows calling `parent()` in your embedded component _slot_. The downside is, there is no concept of a default slot like in @giorgiopogliani's library. Maybe there is a way to achieve but I got stuck in the weeds and gave up :).
